### PR TITLE
Added `MiddlewareExtensions` which provides `addMiddleware` and `addHandler`

### PR DESCRIPTION
**Pull request by kevmoo**
...to `Middleware` instances.
Provides easier access to the composition capabilities offered by `Pipeline`.

### CODE REVIEW

File: lib/src/middleware_extensions.dart [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import 'handler.dart';
+import 'middleware.dart';
+
+/// Extensions on [Middleware] to aid in composing [Middleware] and [Handlers].
+///
+/// These members can be used in place of [Pipeline].
+extension MiddlewareExtensions on Middleware {
+  /// Merges `this` and [other] into a new [Middleware].
+  Middleware addMiddleware(Middleware other) =>
+      (Handler handler) => this(other(handler));
```
Comment (natebosch) [inline lib/src/middleware_extensions.dart:10]: Would nesting this very deeply hurt performance compared to using `Pipeline`? I'm worried about introducing a convenience method if some power users are going to have to steer away from it.

Comment (kevmoo) [reply lib/src/middleware_extensions.dart:10]: I think the perf will be *identical* to what exists now. Look at `Pipeline` – it's the same thing!

Comment (kevmoo) [reply lib/src/middleware_extensions.dart:10]: https://github.com/dart-lang/shelf/blob/a16ea0356c3f73fb9e44973b6a52a06c5546fc75/lib/src/pipeline.dart#L42

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,9 @@
+## 1.2.0
```
Comment (natebosch) [inline CHANGELOG.md:1]:
```suggestion
## 1.2.0-dev
```

**Review by natebosch — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (kevmoo) [reply CHANGELOG.md:1]: I was going to publish right away...

Comment (natebosch) [reply CHANGELOG.md:1]: After it's synced right...

Comment (kevmoo) [reply CHANGELOG.md:1]: Oh...yes. Of course. 🤷

**Pull request merged**